### DISCUSSION TITLE
label propagation from fixed labels

### DISCUSSION THREAD

**Discussion by kkonevets** [networkit/networkit#693] [Q&A]
Hi, 
I try to fix initial labels like here https://igraph.org/r/doc/cluster_label_prop.html

And I could find `baseClustering` parameter in PLP for this, but I don't know how to fill the partition with my predefined labels. I could only find something like `toSingleton()` which doesn't match my task. I should note that some nodes will not have any labels initially, and I want to know them after PLP.

```python
G = readGraph(fname, Format.EdgeListSpaceZero, directed=False)
bc = Partition(size=G.numberOfNodes()) # how can I set initial labels ???

algo = community.PLP(G, baseClustering=bc)
partition = community.detectCommunities(G, algo=algo)
partition.getVector()

```

**Comment by hmeyerhenke** [comment]
The algorithm starts with a singleton clustering. There is no need to set a clustering. Of course, one might want to add this as an optional feature (but please be sure not to break the interface).

**Comment by kkonevets** [comment]
So how can I fix initial labels with `baseClustering`?

**Comment by hmeyerhenke** [comment]
Well, it looks like there is a constructur already for what you have in mind:

PLP::PLP(const Graph& G, const Partition baseClustering, count theta);

Simply pass it a Partition object, the class is in module "structures". For example,
   Partition::Partition(const std::vector<index>& data)
should work.

**Comment by kkonevets** [comment]
Seems like I should implement a custom constructor for python class `networkit.structures.Partition`, which is only `Partition(z=0)` at this moment

**Comment by hmeyerhenke** [comment]
That's quite possible, I only looked at the C++ part.

**Comment by kkonevets** [comment]
thanks

**Comment by kkonevets** [comment]
Just realized that `Partition(const std::vector<index> &data)` sets only elements (nodes), but not labels. And there is no way to map elements to labels (clusters) in this constructor. More over, there is no way to indicate nodes without labels! So the issue should still be open.

**Comment by hmeyerhenke** [comment]
1) The data would be the labels.
2) Modeling the absence of labels is indeed not possible easily in the current implementation.
